src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,320 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
                         {
-                            ReportDiagnostic(context, SpanMissingRule, statements[5].GetLocation(), startToken.Text, endToken.Text);
+                            var argument = ((statements[3] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
+                            ReportDiagnostic(context, SpanMissingRule, argument.GetLocation(), statementIdentifierToken.Text);
                             return false;
                         }
                     }
                     else
                     {
-                        ReportDiagnostic(context, EndSpanMissingRule, statements[4].GetLocation(), openParenToken.Text);
+                        var argument = ((statements[3] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
+                        ReportDiagnostic(context, LocationMissingRule, argument.GetLocation(), statementIdentifierToken.Text);
                         return false;
                     }
                 }
                 else
                 {
-                    ReportDiagnostic(context, StartSpanMissingRule, statements[3].GetLocation(), keywordIdentifierToken.Text);
-                    return false;
-                }
-
-                return true;
-            }
-
-            // Checks the open parenthesis variable, returns a SyntaxToken of "" if analysis failed
-            private SyntaxToken OpenParenAnalysis(SyntaxToken statementIdentifierToken, SyntaxList<StatementSyntax> statements)
-            {
-                var emptyResult = SyntaxFactory.Identifier("");
-
-                var statement = statements[3] as LocalDeclarationStatementSyntax;
-                if (statement == null)
-                {
-                    return emptyResult;
-                }
-
-                SyntaxToken openParenIdentifier = GetIdentifierTokenFromLocalDecl(statement);
-                if (openParenIdentifier == null || openParenIdentifier.Text == "")
-                {
-                    return emptyResult;
-                }
-
-                EqualsValueClauseSyntax equalsValueClause = GetEqualsValueClauseFromLocalDecl(statement);
-                if (equalsValueClause == null)
-                {
-                    return emptyResult;
-                }
-
-                var memberExpression = equalsValueClause.Value as MemberAccessExpressionSyntax;
-                if (memberExpression == null)
-                {
-                    return emptyResult;
-                }
-
-                var identifier = memberExpression.Expression as IdentifierNameSyntax;
-                if (identifier == null || identifier.Identifier.Text != statementIdentifierToken.Text)
-                {
-                    return emptyResult;
-                }
+                    if (statementCount > 9)
```

Comment:
@jepetty I don't think you can get to this point in the code if statementCount >9, so I think this if check is unnecessary